src/sota_tools/ostree_object.cc

```diff
@@ -228,19 +236,22 @@ void OSTreeObject::Upload(const TreehubServer &push_target, CURLM *curl_multi_ha
   curlEasySetoptWrapper(curl_handle_, CURLOPT_WRITEDATA, this);
   http_response_.str("");  // Empty the response buffer
 
-  assert(form_post_ == nullptr);
-  struct curl_httppost *last_item = nullptr;
-  const CURLFORMcode form_err =
-      curl_formadd(&form_post_, &last_item, CURLFORM_COPYNAME, "file", CURLFORM_FILE, file_path_.c_str(), CURLFORM_END);
-  if (form_err != 0u) {
-    // Apparently there is not strerror for formadd.
-    LOG_ERROR << "curl_formadd error: " << form_err;
+  struct stat file_info {};
+  fd_ = fopen(file_path_.c_str(), "rb");
+  if (fd_ == nullptr) {
```

Comment:
As far as I understand, if a file opening or getting its stat fails the code control flow will end up at line #333 ?
I am just wondering if throwing an exception (like at line #229) will work in this case, at first glance it won't work (just thinking loudly :))

Reply:
Throwing an exception might be better.